s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -68,19 +68,17 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
 
   def download(s3Location: S3Location,
                range: Option[ByteRange],
-               sse: Option[ServerSideEncryption]): Source[ByteString, Future[ObjectMetadata]] = {
+               sse: Option[ServerSideEncryption]): (Source[ByteString, NotUsed], Future[ObjectMetadata]) = {
     import mat.executionContext
     val s3Headers = S3Headers(sse.fold[Seq[HttpHeader]](Seq.empty) { _.headersFor(GetObject) })
     val future = request(s3Location, rangeOption = range, s3Headers = s3Headers)
-    Source
-      .fromFuture(future.flatMap(entityForSuccess))
-      .map(_.dataBytes)
-      .flatMapConcat(identity)
-      .mapMaterializedValue { _ =>
-        future.map { resp =>
-          ObjectMetadata(resp.headers)
-        }
-      }
+    (
+      Source
+        .fromFuture(future.flatMap(entityForSuccess))
+        .map(_.dataBytes)
+        .flatMapConcat(identity),
+      future.map(resp ⇒ ObjectMetadata(resp.headers))
+    )
```

Comment:
Took me a moment to see the tuple, wouldn't it be good to use intermediate vals?